homeassistant/components/repetier/sensor.py

```diff
@@ -0,0 +1,294 @@
+"""Support for monitoring Repetier Server Sensors."""
+import logging
+from datetime import timedelta, datetime
+
+from homeassistant.components.repetier import (SENSOR_TYPES)
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.event import track_time_interval
+from homeassistant.util import Throttle
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['repetier']
+
+SCAN_INTERVAL = timedelta(seconds=5)
+TIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the available Repetier Server sensors."""
+    import pyrepetier
+    if discovery_info is None:
+        return
+
+    id = discovery_info['printer']
+    url = discovery_info['url']
+    port = discovery_info['port']
+    apikey = discovery_info['apikey']
+    monitored_conditions = discovery_info['sensors']
+
+    def refresh(event_time):
+        """Update sensors."""
+        _addDevice()
+
+    track_time_interval(hass, refresh, SCAN_INTERVAL)
+
+    def _addDevice():
+        """Add devices."""
+        server = pyrepetier.Repetier(
+            url=url,
+            port=port,
+            apikey=apikey)
+        printers = server.getPrinters()
+
+        devices = []
+
+        for sens_type in monitored_conditions:
+            if sens_type == 'Temperatures':
+                nozzles = server.Nozzle(id)
+                if nozzles is not None:
+                    for nozzle in nozzles:
+                        state = hass.states.get('sensor.' +
```

Comment:
Not all sensors are present in the API when printer is offline or idle.
To avoid creating duplicate sensors I check to see if it is already present or not.

Reply:
Ok. Let's go with my suggestion here to handle this:
https://github.com/home-assistant/home-assistant/pull/21658#discussion_r273398784